node/src/components/storage.rs

```diff
@@ -1,859 +1,578 @@
+    max_deploy_store_size: usize,
+    /// The maximum size of the database to use for the deploy store.
+    ///
+    /// The size should be a multiple of the OS page size.
+    max_deploy_metadata_store_size: usize,
 }
 
-#[allow(trivial_casts)]
-impl<B, D> StorageType for LmdbStorage<B, D>
-where
-    B: Value + WithBlockHeight + 'static,
-    D: Value + Item + 'static,
-{
-    type Block = B;
-    type Deploy = D;
-
-    fn new(config: WithDir<Config>) -> Result<Self> {
-        let root = config.with_dir(config.value().path());
-        fs::create_dir_all(&root).map_err(|error| Error::CreateDir {
-            dir: root.display().to_string(),
-            source: error,
-        })?;
-
-        let block_store_path = root.join(BLOCK_STORE_FILENAME);
-        let block_height_store_path = root.join(BLOCK_HEIGHT_STORE_FILENAME);
-        let deploy_store_path = root.join(DEPLOY_STORE_FILENAME);
-        let chainspec_store_path = root.join(CHAINSPEC_STORE_FILENAME);
-
-        let block_store = LmdbStore::new(block_store_path, config.value().max_block_store_size())?;
-        let block_height_store = LmdbBlockHeightStore::new(
-            block_height_store_path,
-            config.value().max_block_height_store_size(),
-        )?;
-        let deploy_store =
-            LmdbStore::new(deploy_store_path, config.value().max_deploy_store_size())?;
-        let chainspec_store = LmdbChainspecStore::new(
-            chainspec_store_path,
-            config.value().max_chainspec_store_size(),
-        )?;
-
-        Ok(LmdbStorage {
-            block_store: Arc::new(block_store),
-            block_height_store: Arc::new(block_height_store),
-            deploy_store: Arc::new(deploy_store),
-            chainspec_store: Arc::new(chainspec_store),
-        })
-    }
-
-    fn block_store(&self) -> Arc<dyn Store<Value = B>> {
-        Arc::clone(&self.block_store) as Arc<dyn Store<Value = B>>
+impl Default for Config {
+    fn default() -> Self {
+        Config {
+            // No one should be instantiating a config with storage set to default.
+            path: "/dev/null".into(),
+            max_block_store_size: 483_183_820_800,
+            max_deploy_store_size: 322_122_547_200,
+            max_deploy_metadata_store_size: 322_122_547_200,
+        }
     }
+}
 
-    fn block_height_store(&self) -> Arc<dyn BlockHeightStore<B::Id>> {
-        Arc::clone(&self.block_height_store) as Arc<dyn BlockHeightStore<B::Id>>
+impl Config {
+    /// Returns a default `Config` suitable for tests, along with a `TempDir` which must be kept
+    /// alive for the duration of the test since its destructor removes the dir from the filesystem.
+    #[cfg(test)]
+    pub(crate) fn default_for_tests() -> (Self, TempDir) {
+        let tempdir = tempfile::tempdir().expect("should get tempdir");
+        let path = tempdir.path().join("lmdb");
+
+        let config = Config {
+            path,
+            ..Default::default()
```

Comment:
This seems to be cargo culting to me, the impact of changing the number is zero on Linux. What are we hoping to achieve by restricting it?

Reply:
The impact is not zero if tests start failing because we reduce the value.

I believe the objective is to ensure our tests don't fail unexpectedly on CI machines due to lack of disk space.  If we restrict all testing to relatively low values, we can be sure that that isn't a factor when we run on CI machines.  This technique has been in play since before I started here though, so perhaps someone else has a better idea.

I'm not overly strong on this though, unless another reviewer has more details on the issue.

Reply:
As state above, I'm all for abolishing the size limits altogether. Do we move this to a different task?